src/imitation/scripts/train_preference_comparisons.py

```diff
@@ -155,31 +161,55 @@ def train_preference_comparisons(
             reward_net.predict_processed,
             update_stats=False,
         )
-        if agent_path is None:
-            agent = rl_common.make_rl_algo(venv, relabel_reward_fn=relabel_reward_fn)
-        else:
-            agent = rl_common.load_rl_algo_from_path(
-                agent_path=agent_path,
-                venv=venv,
-                relabel_reward_fn=relabel_reward_fn,
-            )
 
-        if trajectory_path is None:
-            # Setting the logger here is not necessary (PreferenceComparisons takes care
-            # of it automatically) but it avoids creating unnecessary loggers.
-            trajectory_generator = preference_comparisons.AgentTrainer(
+        if num_agents < 1 or not isinstance(num_agents, int):
+            raise ValueError("num_agents must be a positive integer!")
+
+        def make_agent_trainer(seed: Optional[int] = None):
+            if agent_path is None:
+                agent = rl_common.make_rl_algo(
+                    venv,
+                    relabel_reward_fn=relabel_reward_fn,
+                )
+            else:
+                agent = rl_common.load_rl_algo_from_path(
```

Comment:
There'll be hilariously little diversity between agents in this case, but not much we can do there. (Support loading *different* agents I guess? But that seems overkill for what's a rare use case.)